src/dataflow/source/kafka.rs

```diff
@@ -42,6 +43,24 @@ lazy_static! {
         &["topic", "source_id", "partition_id"]
     )
     .unwrap();
+    static ref KAFKA_PARTITION_OFFSET_RECEIVED: IntGaugeVec = register_int_gauge_vec!(
+        "mz_kafka_partition_offset_received",
+        "The most recent kafka offset that we have been received by this source",
+        &["topic", "source_id", "partition_id"]
+    )
+    .unwrap();
+    static ref KAFKA_PARTITION_CLOSED_TS: IntGaugeVec = register_int_gauge_vec!(
+        "mz_kafka_partition_closed_ts",
+        "The highest closed timestamp for each partition in this dataflow",
+        &["topic", "source_id", "partition_id"]
+    )
+    .unwrap();
+    static ref KAFKA_CAPABILITY: IntGaugeVec = register_int_gauge_vec!(
+        "mz_kafka_capability",
+        "The current capability for this dataflow",
+        &["topic", "source_id"]
+    )
+    .unwrap();
```

Comment:
Are these just more granular versions of partition_offset_ingested? If that's the case, could you add a comment or alter the help text to explain which one happens before/after the other?

Reply:
OFFSET_RECEIVED happens before OFFSET_INGESTED.

The rest have to do with timestamps and not offsets.